packages/completer/src/tokens.ts

```diff
@@ -49,6 +49,19 @@ export interface ICompletionProvider<
    */
   readonly identifier: string;
 
+  /**
+   * Rank use to order completion results from different completion providers.
+   *
+   * #### Note: The default providers (CompletionProvider:context and
+   * CompletionProvider:kernel) use a rank of ≈500. If you want to give
+   * priority to your provider, use a rank of 1000 or above.
```

Comment:
I agree with the approach in general but instead of using rank 500 and 501 maybe we should leave some space between them to allow providers higher than one but lower than the other? For example 500 and 550